### PR TITLE
Feat/single session

### PR DESCRIPTION
**Pull request by martinhaex**
@jrgns single session

### CODE REVIEW

File: lib/ditty/listener.rb [hunk 1]
```diff
@@ -50,9 +52,49 @@ def action_from(target, method)
...
-      @mutex.synchronize { Ditty::AuditLog.create values }
+      require 'ditty/models/audit_log'
+
+      values[:user] ||= values[:target].current_user if values[:target] && values[:target].respond_to?(:current_user)
+      values.delete(:active) if values.include? :active
+      @mutex.synchronize { ::Ditty::AuditLog.create values }
+    end
+
+    def user_login(event)
+      #TODO: Check for existing active sessions and marks them inactive
```
Comment (jrgns) [inline lib/ditty/listener.rb:63]: TODO? Is this for future changes?

File: lib/ditty/controllers/application.rb [hunk 2]
```diff
@@ -169,6 +174,21 @@ def view_location
...
         content_type(:json) if request.accept.count.eql?(1) && request.accept.first.to_s.eql?('*/*')
       end
+      
+      return unless respond_to?(:browser) &&
+                    respond_to?(:request) &&
+                    respond_to?(:current_user) &&
+                    current_user.is_a?(Ditty::User) &&
+                    request.path_info != '/auth/identity/callback'
+
+      active_trait = current_user.user_login_traits.select { |t| t.active == true }.first
```
Comment (jrgns) [inline lib/ditty/controllers/application.rb:184]: Any way we can turn this feature on and off using a setting?

File: lib/ditty/listener.rb (hunk 1 above)
Comment (martinhaex) [reply lib/ditty/listener.rb:63]: No I need to remove that..

File: lib/ditty/controllers/application.rb (hunk 2 above)
Comment (martinhaex) [reply lib/ditty/controllers/application.rb:184]: No, I'll code it in.

**Pull request merged**